src/components/SearchBar.tsx

```diff
@@ -20,50 +20,61 @@ const searchConfig = {
   },
 };
 
-const SearchBar = forwardRef<HTMLInputElement, Props>(({ buttonType = "cancel", onClick }, ref) => {
-  const [isFocused, setIsFocused] = useState(true);
-  const searchDiv = useRef<HTMLInputElement>(null);
+const SearchBar = forwardRef<HTMLInputElement, Props>(
+  ({ buttonType = "cancel", onClick }, ref) => {
+    const [isFocused, setIsFocused] = useState<boolean>(false);
+    const [isFilled, setIsFilled] = useState<boolean>(false);
+    const searchDiv = useRef<HTMLInputElement>(null);
 
-  const handleFocus = () => {
-    searchDiv.current?.focus();
-    setIsFocused(!isFocused);
-  };
+    const onFocus = () => {
+      searchDiv.current?.focus();
+      setIsFocused(!isFocused);
+    };
 
-  const handleBlur = () => {
-    searchDiv.current?.blur();
-    setIsFocused(!isFocused);
-  };
+    const onBlur = () => {
+      searchDiv.current?.blur();
+      setIsFocused(!isFocused);
+    };
 
-  return (
-    <div className="flex bg-tertiary-700 w-full px-4 py-3">
-      <div ref={searchDiv} className="opacity-50 bg-tertiary-500 rounded-md">
+    const onChange = () => {
+      setIsFilled(!isFilled);
+    };
+
+    return (
+      <div className="flex bg-tertiary-700 w-full px-4 py-3">
         <div
-          className="flex items-center p-2 border-primary-400"
-          onFocus={handleFocus}
-          onBlur={handleBlur}
+          ref={searchDiv}
+          onFocus={onFocus}
+          onBlur={onBlur}
+          className={`bg-tertiary-500/50 rounded-md ${
+            isFocused ? "border border-primary-400" : ""
+          }`}
```

Comment:
```suggestion
          className={`bg-tertiary-500/50 rounded-md ${
            isFocused ? "ring-inset ring-1 ring-primary-400" : ""
          }`}
```

이렇게 작성하시면 border를 안쪽 영역으로 먹일 수 있습니당.
border로 한 후에 inset을 해주는 방법도 있는 것 같긴 한데... 일단은 이렇게 하면 잘 될거에요!